SingularityBase/src/main/java/com/hubspot/singularity/api/SingularityDeployRequest.java

```diff
@@ -12,15 +13,22 @@
   private final Optional<Boolean> unpauseOnSuccessfulDeploy;
   private final SingularityDeploy deploy;
   private final Optional<String> message;
+  private final Optional<SingularityRequest> newRequestData;
 
   @JsonCreator
   public SingularityDeployRequest(
       @JsonProperty("deploy") SingularityDeploy deploy,
       @JsonProperty("unpauseOnSuccessfulDeploy") Optional<Boolean> unpauseOnSuccessfulDeploy,
-      @JsonProperty("message") Optional<String> message) {
+      @JsonProperty("message") Optional<String> message,
+      @JsonProperty("newRequestData") Optional<SingularityRequest> newRequestData) {
```

Comment:
I had it in the deploy initially. But, the usage of this new field would only be at deploy time. After the deploy fails or succeeds we are back to using the normal request data. Having request data that stuck around with the deploy seemed like it could be more confusing. Made more sense to put it outside the deploy so it didn't stick around